### PR TITLE
chore: Components sizing and alignment

### LINKED ISSUE #30426: Components sizing and alignment
As part of the task, we need to add small size for the components
- TextInput
- Button
- IconButton
- ActionGroup

It is also necessary to align all the above components. 

Additional info:
https://theappsmith.slack.com/archives/C04P60V0VQE/p1700233776321719?thread_ts=1700233527.286549&cid=C04P60V0VQE
https://www.notion.so/appsmith/Alignment-and-size-issues-b7fcbe8420164b04b0b40620c82a416f?pvs=4#3baaa44a6a894ef59b598ee31f6d62be
Desired result: button with icon should be the same height as icon with text, it should be of square proportions. We know corresponding cap-size of the text and inner spacing used in a button. Width can be then set with aspect-ratio: 1.

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #30426 

## Summary by CodeRabbit

- **New Features**
  - Enhanced layout configuration for various widgets including Button, Checkbox, Currency Input, Icon Button, Input, Menu Button, Paragraph, Phone Input, Radio Group, Switch, and Table, providing better responsiveness and design consistency.

- **Bug Fixes**
  - Fixed the `componentHeight` calculation in the Table widget for improved stability.

- **Refactor**
  - Streamlined auto layout configuration across multiple widgets to simplify the user interface.

- **Style**
  - Updated default widget dimensions for a more cohesive user experience.

- **Chores**
  - Cleaned up unused properties and imports to optimize widget performance.

### CODE REVIEW

File: app/client/packages/design-system/widgets/src/components/Icon/src/styles.module.css [hunk 1]
```diff
@@ -1,16 +1,15 @@
 .icon {
+  aspect-ratio: 1;
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Icon/src/styles.module.css:2]: Does this property affect something?

File: app/client/packages/design-system/widgets/src/components/ActionGroup/src/types.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export interface ActionGroupProps<T>
     >,
     InheritedActionButtonProps {
   orientation?: keyof typeof ACTION_GROUP_ORIENTATIONS;
+  size?: "small" | "medium";
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/ActionGroup/src/types.ts:35]: Let's make a shared interface for size, as we do with [colors](https://github.com/appsmithorg/appsmith/blob/release/app/client/packages/design-system/widgets/src/shared/colors/index.ts). This way we can maintain consistency in the naming of the props.

File: app/client/packages/design-system/widgets/src/components/Menu/stories/Menu.stories.mdx [hunk 3]
```diff
@@ -61,20 +58,9 @@ Item props are not pulled up in the ArgsTable, the data can be found [here](http
...
-      </Item>
-      <Item
-        key="delete"
-        icon={DeleteBinLineIcon}
-        color="negative"
-        iconPosition="end"
-      >
-        Delete
-      </Item>
+      <Item key="copy">{"Copy"}</Item>
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Menu/stories/Menu.stories.mdx:61]: The menu with the icon no longer works. Seems due to changes in the Item component.

Снимок экрана 2024-01-25 в 10 24 40

File: app/client/packages/design-system/widgets/src/components/Icon/src/styles.module.css [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
+
   &[data-size="small"] {
     inline-size: 1rem;
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Icon/src/styles.module.css:5]: As I understand it, there should be fixed values for sizes. Apparently, we need a separate token that responds to changes in the topic. You also need a separate token to control stroke.

File: app/client/packages/design-system/widgets/src/components/Menu/stories/Menu.stories.mdx [hunk 5]
```diff
@@ -1,9 +1,6 @@
+import { COLORS } from "@design-system/widgets";
 import { Button, Menu, MenuList, Item } from "@design-system/widgets";
 import { Canvas, Meta, Story, ArgsTable } from "@storybook/addon-docs";
-import { COLORS } from "@design-system/widgets";
-import FileCopyLineIcon from "remixicon-react/FileCopyLineIcon";
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Menu/stories/Menu.stories.mdx:4]: The last remixicon icon in the headless checkbox remains, it also needs to be replaced.

**Review by KelvinOm — CHANGES_REQUESTED** [review]

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css [hunk 6]
```diff
@@ -161,29 +158,31 @@
   &[data-icon-button] {
     text-align: center;
     padding: 0;
-    container-type: inline-size;
+    aspect-ratio: 1;
```
[the remark this answers is not in the document]
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:161]: previously we were setting sizes of icons with container units.

File: app/client/packages/design-system/widgets/src/components/Icon/src/styles.module.css (hunk 1 above)
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/Icon/src/styles.module.css:2]: Yes. For icons, we are setting the width and this property just sure the height is also same.

**Review by ichik — APPROVED** [review]

File: app/client/packages/design-system/widgets/src/components/Menu/stories/Menu.stories.mdx (hunk 5 above)
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/Menu/stories/Menu.stories.mdx:4]: changed the svg with tabler icons's svg.

File: app/client/packages/design-system/headless/src/components/Checkbox/src/icons/SubtractIcon.tsx [hunk 7]
```diff
@@ -4,8 +4,19 @@ import type { ComponentProps } from "react";
 // Adapted from remixicon-react/SubtractLineIcon (https://github.com/Remix-Design/RemixIcon/blob/f88a51b6402562c6c2465f61a3e845115992e4c6/icons/System/subtract-line.svg)
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Checkbox/src/icons/SubtractIcon.tsx:4]: The comment is not relevant anymore.

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css (hunk 6 above)
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:161]: I just don't understand what it's affecting right now.

https://github.com/appsmithorg/appsmith/assets/11555074/b120265e-fd31-4ab5-b2c2-28b34c10cd3c

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**